sqlglot/optimizer/qualify_tables.py

```diff
@@ -163,5 +174,13 @@ def _qualify(table: exp.Table) -> None:
                         column.set(p, None)
 
                     column.set("table", table_alias.copy())
+            elif (
+                canonical_aliases
+                and table
+                and (canonical_table := canonical_aliases.get(table, "")) != column.table
+            ):
+                # Amend existing aliases, e.g. t.c -> _0.c if t is aliased to _0
+                column.set("table", exp.to_identifier(canonical_table))
+                pass
```

Comment:
you don't need this pass right?

Reply:
Discussed this in Slack. The conclusion was that there exists a bug in main today too, where we incorrectly mutate multi-part column references, without checking whether they represent struct field lookups (i.e., `Dot` sequences).

We should look into this as a follow-up to this PR, to avoid increasing the current scope.